py/lexer.c

```diff
@@ -688,26 +688,18 @@ mp_lexer_t *mp_lexer_new(qstr src_name, mp_reader_t reader) {
     // store sentinel for first indentation level
     lex->indent_level[0] = 0;
 
-    // preload characters
-    lex->chr0 = reader.readbyte(reader.data);
+    // load lexer with start of file, advancing lex->column to 1
+    lex->chr0 = 0;    // dummy character burned off by next_char()
```

Comment:
You could make this a '\n', and initialise lex->line to 0, and then next_char() would handle correctly the case of an input stream of 0 chars.  Then below you just need to handle the case of an input stream of 1 char.